File: python/54c_1873D_1DEraser.py
def solve(s, n, k):
    cnt, i = 0, 0
    while i < n:
        if s[i] == "B":
            cnt += 1
            i += k
        else:
            i += 1
    return cnt

def main():
    ans = []
    t = int(input())
    for _ in range(t):
        n, k = input().split()
        n, k = int(n), int(k)
        s = input().strip()
        ans.append(solve(s, n, k))
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()

File: cpp/54c_1873D_1DEraser.cpp
#include<bits/stdc++.h>
int solve(std::string s, int n, int k){
    int cnt = 0, i = 0;
    while (i < n){
        if(s[i] == 'B'){
            cnt++;
            i = i + k;
        }
        else{
            i++;
        }
    }
    return cnt;
}

int main(){
    int t, n, k;
    std::string s;
    std::vector<int> ans;
    std::cin>>t;
    for (int i = 0; i < t; i++){
        std::cin>>n>>k;
        std::cin>>s;
        ans.push_back(solve(s, n, k));
    }
    for (auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}